packages/react-events/src/dom/shared/index.js

```diff
@@ -70,3 +70,17 @@ export function hasModifierKey(event: ReactDOMResponderEvent): boolean {
     altKey === true || ctrlKey === true || metaKey === true || shiftKey === true
   );
 }
+
+// Keyboards, Assitive Technologies, and element.click() all produce "virtual"
+// clicks that do not include coordinates and "detail" is always 0 (where
+// pointer clicks are > 0).
+export function isVirtualClick(event: ReactDOMResponderEvent): boolean {
+  const nativeEvent: any = event.nativeEvent;
+  return (
+    nativeEvent.detail === 0 &&
+    nativeEvent.screenX === 0 &&
+    nativeEvent.screenY === 0 &&
+    nativeEvent.clientX === 0 &&
+    nativeEvent.clientY === 0
+  );
+}
```

Comment:
(Not sure what the etiquette on a merged PR is -- please let me know if these comments would be more appropriate in e.g. a new Issue.)

I don't believe this works cross-browser.

1) in IE(11), detail is always `0`, even for mouse-initiated click events. Can check for `!nativeEvent.pointerType` to work around this, since IE11 dispatches PointerEvents for these.

2) in Safari, the screen and client coordinates are present even for a keyboard-initiated click. The client coordinates seem to default to the "center" of the element.

What I've used internally at Tableau for this that works cross-browser is `return (nativeEvent.detail === 0 && !nativeEvent.pointerType);`

Reply:
This is really awesome feedback. Thank you cc @necolas seems like a valid thing we should be doing too.

Reply:
How does `(nativeEvent.detail === 0 && !nativeEvent.pointerType)` work if that evaluates as `true` for mouse events as well as virtual clicks?

Reply:
My investigation showed that `nativeEvent.detail === 0` is sufficient for all evergreen browsers; adding `!nativeEvent.pointerEvent` covers the IE11 case. Is that more clear?

Reply:
I still don't follow. The `click` event is of type `MouseEvent` which has no `pointerType` property. Are you saying IE11's `click` event is of type `PointerEvent`?

Reply:
> Are you saying IE11's click event is of type PointerEvent?
Yes, exactly

Reply:
Thanks for confirming; we didn't know about that quirk. We'll be sure to incorporate your suggested fix. I like what Safari does here. I'm also curious to hear what you use this check for at Tableau

Reply:
https://github.com/facebook/react/pull/16915

Reply:
> I'm also curious to hear what you use this check for at Tableau

It's a long, annoying story -- the short version is there an event normalizer (parallel to React) sitting on top of touch events that interferes with native keyboard click events on button elements. I needed this to differentiate the event sources so it could ignore those from the keyboard.

Reply:
@necolas little bit of unfortunate news -- it appears this will incorrectly eval to `true` for click events which are dispatched to an `<input>` when clicked on a wrapping `<label>`. Not sure if this is a problem for your purposes, but wanted to give you a heads-up.